xarray/core/accessors.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from .common import is_datetime_like
+from .extensions import register_dataarray_accessor
+from .pycompat import dask_array_type
+
+import numpy as np
+import pandas as pd
+
+
+def _get_date_field_from_dask(values, name):
+    """Specialized date field accessor for data contained in dask arrays
+    """
+    from dask import delayed
+    from dask.array import from_delayed, map_blocks
+    from dask.dataframe import from_array
+
+    @delayed
+    def _getattr_from_dask(accessor):
+        attr_values = getattr(accessor, name).values
+        return attr_values.compute()
+
+    def _ravel_and_access(darr):
+        raveled = darr.ravel()
+        raveled_as_series = from_array(raveled[..., np.newaxis],
+                                       columns=['_raveled_data', ])
+        field_values = _getattr_from_dask(raveled_as_series['_raveled_data'].dt)
+        field_values = from_delayed(field_values, shape=raveled.shape,
+                                    dtype=raveled.dtype)
+        return field_values.reshape(darr.shape).compute()
+
+    return map_blocks(_ravel_and_access, values, dtype=values.dtype)
```

Comment:
`dtype` needs to be the dtype of the return value, not the input. For most of these accessors, this is just `np.int64` (for strings, use `dtype=object`).